### PR TITLE
Database connector

### PR DESCRIPTION
**Pull request by BlankAmber**
Implements the following:
1. Runs linter and tests independent of mvn clean install
- This change is neccessary for the Database connector PR as this PR is the first one to add proper functionality (requires unit tests which should be run automatically through GitHub Actions)
- Originally "Build maven" would run the checkstyle checks as well which is a bit misleading
- Now they are separate, `mvn test` is now also being run after the checkstyle checks (Note: no tests have actually been implemented yet)
- Old: Screenshot 2023-12-06 at 09 49 42
- New: 
Screenshot 2023-12-06 at 09 51 41
2. Adjusts linter config
3. Adds the database connector + appropriate unit tests

To use this locally remember to set environment variables for DB_USERNAME, DB_PASSWORD, DB_HOST, and DB_NAME.

E.g.
`export VAR_NAME="value here"`

Also set the environment variables for the run configuration in IntelliJ too.

### CODE REVIEW

**Comment by BlankAmber** [comment]
TODO: Check if 04b2409 is actually necessary

**Review by adammarshall745 — APPROVED** [review]
This will be very usful, thank you

File: src/main/java/org/kainos/ea/db/DatabaseConnector.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    public static final String ENV_DB_HOST = "DB_HOST";
+    public static final String ENV_DB_NAME = "DB_NAME";
+
+    private static Connection connection;
+
+    public static void resetConnection() {
+        DatabaseConnector.connection = null;
+    }
+
+    private DatabaseConnector() {
```
Comment (lullabelle) [inline src/main/java/org/kainos/ea/db/DatabaseConnector.java:19]: What is this for?

**Review by lullabelle — APPROVED** [review]
Approved. Please can you tell me why the empty database connection constructor?

File: src/main/java/org/kainos/ea/db/DatabaseConnector.java (hunk 1 above)
Comment (BlankAmber) [reply src/main/java/org/kainos/ea/db/DatabaseConnector.java:19]: > What is this for?

I'm using DatabaseConnector like a static utility class so it's not supposed to be instantiated (the linter I'm using suggested adding an empty private constructor)

**Comment by BlankAmber** [comment]
> Approved. Please can you tell me why the empty database connection constructor?

I'm using DatabaseConnector like a static utility class so it's not supposed to be instantiated (the linter I'm using suggested adding an empty private constructor)

**Pull request merged**